pulsar/producer.go

```diff
@@ -76,6 +76,15 @@ type ProducerOptions struct {
 	// This properties will be visible in the topic stats
 	Properties map[string]string
 
+	// SendTimeout set the timeout for a message that not be acknowledged by server since sent.
+	// Send and SendAsync returns an error after timeout.
+	// Default is 30 seconds, -1 to disable.
+	SendTimeout time.Duration
```

Comment:
Shouldn't this be taken the `ctx` passed in by the application at the moment of `Send()` ?

Reply:
Thanks for review.
I see [PR#252](https://github.com/apache/pulsar-client-go/pull/252) implementing `sendTimeout` with context but progress has stalled.
I'm going to try context instead of timer.